modules/jpms-utils/src/main/java11/org/mule/runtime/jpms/api/JpmsUtils.java

```diff
@@ -283,8 +333,9 @@ public static ModuleLayer createModuleLayer(URL[] modulePathEntries, ClassLoader
                                                 .collect(toList()));
       controller = defineModulesWithOneLoader(configuration,
                                               asList(isolatedModulesController.layer(), resolvedParentLayer),
-                                              parentLayer.map(layer -> layer.findLoader(layer.modules()
-                                                  .iterator().next().getName())).orElse(parent));
+                                              parentClassLoaderResolver.apply(parentLayer
```

Comment:
this layer shouldn't have this process, it is already applied to its parent layer (the one from `isolatedModulesController`)